src/Makefile

```diff
@@ -17,6 +18,54 @@ IOS_BUILD_DIR=$(BUILD_DIR)/ios
 WATCH_BUILD_DIR=$(BUILD_DIR)/watch
 TVOS_BUILD_DIR=$(BUILD_DIR)/tvos
 
+MACOS_DOTNET_BUILD_DIR=$(DOTNET_BUILD_DIR)/macos
+IOS_DOTNET_BUILD_DIR=$(DOTNET_BUILD_DIR)/ios
+WATCHOS_DOTNET_BUILD_DIR=$(DOTNET_BUILD_DIR)/watchos
+TVOS_DOTNET_BUILD_DIR=$(DOTNET_BUILD_DIR)/tvos
+
+GENERATOR_FLAGS=-process-enums -core -nologo -nostdlib -noconfig -native-exception-marshalling --ns:ObjCRuntime
+
+DOTNET_FLAGS=/noconfig /nostdlib+ /deterministic /features:strict /nologo /target:library /debug /unsafe /define:DOTNET /define:DOTNET_TODO \
+	/r:$(DOTNET_BCL_DIR)/mscorlib.dll \
```

Comment:
This doesn't sound right. There should be no `mscorlib`, it is `System.Private.CoreLib` and it should not be referenced directly. Last time I tried the assemblies needed were only those:
- System.Collections.dll
- System.Drawing.Primitives.dll
- System.Linq.dll
- System.Net.Http.dll
- System.Runtime.dll
- System.Runtime.InteropServices.dll
- System.Security.Cryptography.X509Certificates.dll
- System.Xml.dll
- System.Xml.ReaderWriter.dll

Reply:
mscorlib.dll is there:

```shell
$  ls -la /usr/local/share/dotnet/packs/Microsoft.NETCore.App.Ref/3.1.0/ref/netcoreapp3.1/mscorlib.dll 
-rw-r--r--  1 root  wheel  55880 Nov 15 09:36 /usr/local/share/dotnet/packs/Microsoft.NETCore.App.Ref/3.1.0/ref/netcoreapp3.1/mscorlib.dll
```

that said, I did not try to minimize the list of assemblies, I found a set that worked, and went with that. I'll try your list to see if that works.

Reply:
It is there as a facade for compatibility with .NET Framework libraries, not for consumption from new code.

Reply:
I wasn't able to remove many assemblies from the list, I had to add most of them back when I tried your list of assemblies. This is what happened after the initial removal and after adding them back one by one: https://gist.github.com/rolfbjarne/7f515ae2ac7c0f712b3d70f0ff985549

Reply:
Interesting. I'll check later what I did differently, perhaps I disabled OpenTK and used slightly different build options. Nevertheless, as long as it doesn't depend on compatibility facade assemblies I am happy. 👍